common/djangoapps/third_party_auth/tests/test_views.py

```diff
@@ -153,3 +155,42 @@ def test_login_disabled(self):
         self.enable_saml(enabled=False)
         response = self.client.get(self.LOGIN_URL)
         self.assertEqual(response.status_code, 404)
+
+
+@unittest.skipUnless(AUTH_FEATURE_ENABLED, AUTH_FEATURES_KEY + ' not enabled')
+class IdPRedirectViewTest(SAMLTestCase):
+    """
+        Test IdPRedirectView.
+    """
+    def setUp(self):
+        super(IdPRedirectViewTest, self).setUp()
+
+        self.enable_saml()
+        self.configure_saml_provider(
+            name="Test",
+            slug="test",
+            enabled=True,
+        )
+
+    def test_with_valid_provider_slug(self):
+        endpoint_url = self.get_idp_redirect_url('saml-test')
+        expected_url = pipeline.get_login_url('saml-test', pipeline.AUTH_ENTRY_LOGIN, reverse('dashboard'))
+
+        response = self.client.get(endpoint_url)
+
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.url, expected_url)
+
+    def test_with_invalid_provider_slug(self):
+        endpoint_url = self.get_idp_redirect_url('saml-test-invalid')
+
+        response = self.client.get(endpoint_url)
+
+        self.assertEqual(response.status_code, 404)
+
+    @staticmethod
```

Comment:
Why staticmethod?

Reply:
I thought that this was more of a utility function and wasn't tied to the actual instance of the test class. My understanding was it was appropriate to use a static method in Python when you don't need access to the attributes of an instance of a class but want to group methods within the class because they are logically related. What do you think?